exchanges/poloniex/poloniex_wrapper.go

```diff
@@ -145,132 +163,234 @@ func (e *Exchange) SetDefaults() {
 
 // Setup sets user exchange configuration settings
 func (e *Exchange) Setup(exch *config.Exchange) error {
-	err := exch.Validate()
-	if err != nil {
+	if err := exch.Validate(); err != nil {
 		return err
 	}
 	if !exch.Enabled {
 		e.SetEnabled(false)
 		return nil
 	}
-	err = e.SetupDefaults(exch)
-	if err != nil {
+	if err := e.SetupDefaults(exch); err != nil {
 		return err
 	}
 
-	wsRunningURL, err := e.API.Endpoints.GetURL(exchange.WebsocketSpot)
+	if err := e.Websocket.Setup(&websocket.ManagerSetup{
+		ExchangeConfig: exch,
+		FillsFeed:      e.Features.Enabled.FillsFeed,
+		TradeFeed:      e.Features.Enabled.TradeFeed,
+		Features:       &e.Features.Supports.WebsocketCapabilities,
+		OrderbookBufferConfig: buffer.Config{
+			SortBuffer:            true,
+			SortBufferByUpdateIDs: true,
+		},
+		UseMultiConnectionManagement: true,
+	}); err != nil {
+		return err
+	}
+	wsSpot, err := e.API.Endpoints.GetURL(exchange.WebsocketSpot)
 	if err != nil {
 		return err
 	}
-
-	err = e.Websocket.Setup(&websocket.ManagerSetup{
-		ExchangeConfig:        exch,
-		DefaultURL:            poloniexWebsocketAddress,
-		RunningURL:            wsRunningURL,
-		Connector:             e.WsConnect,
+	if err := e.Websocket.SetupNewConnection(&websocket.ConnectionSetup{
+		ResponseCheckTimeout:  exch.WebsocketResponseCheckTimeout,
+		ResponseMaxLimit:      exch.WebsocketResponseMaxLimit,
+		URL:                   wsSpot,
+		RateLimit:             request.NewWeightedRateLimitByDuration(500 * time.Millisecond),
 		Subscriber:            e.Subscribe,
 		Unsubscriber:          e.Unsubscribe,
-		GenerateSubscriptions: e.GenerateDefaultSubscriptions,
-		Features:              &e.Features.Supports.WebsocketCapabilities,
-		OrderbookBufferConfig: buffer.Config{
-			SortBuffer:            true,
-			SortBufferByUpdateIDs: true,
-		},
-	})
+		GenerateSubscriptions: e.generateSubscriptions,
+		Handler:               e.wsHandleData,
+		Connector:             e.wsConnect,
+		MessageFilter:         connSpotPublic,
+	}); err != nil {
+		return err
+	}
+	wsSpotPrivate, err := e.API.Endpoints.GetURL(exchange.WebsocketPrivate)
+	if err != nil {
+		return err
+	}
+	if err := e.Websocket.SetupNewConnection(&websocket.ConnectionSetup{
+		ResponseCheckTimeout:  exch.WebsocketResponseCheckTimeout,
+		ResponseMaxLimit:      exch.WebsocketResponseMaxLimit,
+		URL:                   wsSpotPrivate,
+		RateLimit:             request.NewWeightedRateLimitByDuration(500 * time.Millisecond),
+		Subscriber:            e.Subscribe,
+		Unsubscriber:          e.Unsubscribe,
+		GenerateSubscriptions: e.generatePrivateSubscriptions,
+		Handler:               e.wsHandleData,
+		Connector:             e.wsAuthConn,
+		MessageFilter:         connSpotPrivate,
+		Authenticated:         true,
```

Comment:
Authenticated bool field is not used for multiconnection as it assumes authentication is available via the function field `Authenticate`